src/qir2qasm_trans/qir_trans/qir_profile.py

```diff
@@ -0,0 +1,191 @@
+from typing import Dict, List, Union
+
+from llvmlite.ir import (
+    Context,
+    DoubleType,
+    FunctionType,
+    IdentifiedStructType,
+    IntType,
+    Type,
+    VoidType,
+)
+from openqasm3 import ast
+
+from .builder import (
+    BinaryExpressionBuilder,
+    ConstantBuilder,
+    DeclBuilder,
+    FunctionBuilder,
+    FunctionInfo,
+    GateBuilder,
+    InputBuilder,
+    InstructionBuilder,
+    InstructionInfo,
+    InttoptrBuilder,
+    LoadBuilder,
+    MeasurementBuilder,
+    OutputBuilder,
+    QubitDeclarationBuilder,
+    RecordBuilder,
+    ResetBuilder,
+    ResultDeclarationBuilder,
+    StructureInfo,
+    build_rotation_2Q_definition,
+)
+
+
+class Profile:
+    def __init__(self, name: str):
+        self.name = name
+        self.structures: Dict[str, StructureInfo] = {}
+        self.classical_instruction: Dict[str, InstructionInfo] = {}
+        self.standard_functions: Dict[str, FunctionInfo] = {}
+        self.context = Context()
+
+        self._define_structs()
+        self._define_functions()
+        self._define_classical_instructions()
+
+    def register_function(
+        self,
+        func_name: str,
+        ret_type: Type,
+        arg_types: List[Type],
+        def_statement: ast.QuantumGateDefinition,
+        func_builder: FunctionBuilder,
+    ):
+        fn_type = FunctionType(ret_type, arg_types)
+        self.standard_functions[func_name] = FunctionInfo(fn_type, def_statement, func_builder)
+
+    def register_classical_instruction(self, name: str, inst_builder: InstructionBuilder):
+        self.classical_instruction[name] = InstructionInfo(builder=inst_builder)
+
+    def register_structure(
+        self,
+        name: str,
+        type_qir: IdentifiedStructType,
+        type_ast: Union[ast.ClassicalType, str],
+        decl_builder: DeclBuilder,
+    ):
+        # self.structures[name] = StructureInfo(type_qir, None, decl_builder)
+        self.structures[name] = StructureInfo(type_qir, type_ast, decl_builder)
+
+    def _define_structs(self):
+        pass
+
+    def _define_functions(self):
+        pass
+
+    def _define_classical_instructions(self):
+        pass
+
+
+class BaseProfile(Profile):
+    def __init__(self):
+        super().__init__("base_profile")
+
+    def _define_structs(self):
+        self.register_structure(
+            "Qubit", self.context.get_identified_type("Qubit"), "Qubit", QubitDeclarationBuilder()
```

Comment:
What are other type that you expect to be in near term quantum programs? This is not something users will arbitrarily expand, right?

If so, we should be define an `Enum` class on what types are allowed. This makes it more clear for users and developer about the expected behavior in types